### PR TITLE
core/bpf-firewall: make bpf_firewall_supported() always set unsupport…

### PR DESCRIPTION
**Pull request by yuwata**
…ed reason when BPF_FIREWALL_UNSUPPORTED is returned

Otherwise, log_unit_full_errno() in emit_bpf_firewall_warning() will
trigger an assertion.

This is a fix for non-test mode. For test mode, see #21583.

### CODE REVIEW

**Comment by yuwata** [comment]
Now the negative errno is propagated.

**Comment by bluca** [comment]
There's still a return path where `bpf_firewall_unsupported_reason` is not set? Just after the invocation of `cg_unified_controller()`

**Comment by yuwata** [comment]
> There's still a return path where `bpf_firewall_unsupported_reason` is not set? Just after the invocation of `cg_unified_controller()`

The function is called as

main() -> manager_new() -> manager_setup_cgroup() -> cg_bpf_mask_supported() -> bpf_firewall_supported() -> cg_unified_controller()

And now the failure is propagated. So on failure, `emit_bpf_firewall_warning()` is not called.

**Review by bluca — APPROVED** [review]

File: src/core/cgroup.c [hunk 1]
```diff
@@ -3228,26 +3228,36 @@ static int cg_bpf_mask_supported(CGroupMask *ret) {
 
         /* BPF-based firewall */
         r = bpf_firewall_supported();
+        if (r < 0)
```
Comment (anitazha) [inline src/core/cgroup.c:3231]: @yuwata do you think there's a risk of people adding a new "is supported" function that returns `r < 0` for minor errors?

The way it's written now only fatal errors should return `r < 0` for "is supported" functions. But I'm concerned someone will accidentally return `r < 0` in the future for something like `EPERM`. I feel like the way it was before this was added was less error-prone. What do you think?

Comment (yuwata) [reply src/core/cgroup.c:3231]: @anitazha Sorry for late response.
> do you think there's a risk of people adding a new "is supported" function that returns r < 0 for minor errors?

In that case, manage_new() fails, and we failed to boot system correctly. So, I hope such failure is detected CIs and then we notice the direct regression.

But for safety, handling that `bpf_xxx_supported()` reported a negative errno as `not supported` may be also a good idea.
If you'd like, please submit a PR.

**Pull request merged**